core/string/ustring.cpp

```diff
@@ -4352,9 +4352,22 @@ bool String::is_valid_filename() const {
 		return false;
 	}
 
+	// TODO: Use invalid_filename_characters
 	return !(find(":") != -1 || find("/") != -1 || find("\\") != -1 || find("?") != -1 || find("*") != -1 || find("\"") != -1 || find("|") != -1 || find("%") != -1 || find("<") != -1 || find(">") != -1);
```

Comment:
Would be good to solve the TODO now, shouldn't be hard.

One solution would be `return this == this->validate_filename();` but it's likely more expensive than checking char by char, and it would be good to keep this as fast as possible.

So I guess (untested):
```cpp
Vector<String> chars = String(invalid_filename_characters).split(" ");
for (const String &char : chars) {
	if (contains(char)) {
		return false;
	}
}
return true;
```

Reply:
okay, done. We do seem to have a unit test for this.